iroh/src/endpoint.rs

```diff
@@ -3054,6 +3057,81 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    #[traced_test]
+    async fn can_connect_from_same_id() -> Result {
+        std::env::set_var("RUST_LOG", "warn");
+        const ECHOS: usize = 3;
+
+        let server = Endpoint::builder()
+            .alpns(vec![TEST_ALPN.to_vec()])
+            .discovery_local_network()
```

Comment:
I wonder if this test really needs a discovery service.  Would it not work anyway because both endpoints are on the same network, so the NodeAddr should contain a usable IP addr.